subsys/shell/shell_utils.c

```diff
@@ -313,10 +313,7 @@ u16_t shell_str_similarity_check(const char *str_a, const char *str_b)
 		if (str_a[cnt] != str_b[cnt]) {
 			return cnt;
 		}
-
-		if (++cnt == 0) {
-			return --cnt; /* too long strings */
-		}
```

Comment:
This part of code cannot be removed as it prevents from stucking in while loop.

Reply:
It is when `cnt` overflows.

Reply:
On the second thought I would recommend to change it to:

`if (++cnt == 0U)`

Reply:
wow, you are considering the overflow here. Ok, this is nasty :) If you are considering it this function is wrong IMHO since we are returning a value that the user canw interpret as the number of equal chars and that after that the strings are different.
If an overflow happens this function should return an error since the value. The other thing is using a properly type for string length like size_t, that is the value returned by string length.

Reply:
@ceolin : I have digged a bit deeper :) It seems that this function is no longer in use, it  has been replaced by `shell_str_common`. So the whole function can be removed from .c and .h files. Would you mind to update this PR?

Reply:
nope, I'll do it :)